src/components/LoginDialog.jsx

```diff
@@ -15,9 +15,9 @@ export default function LoginDialog() {
 
   const navigate = useNavigate();
   const closeDialog = () =>
-    navigate('', {
+    navigate(location.pathname, {
       replace: true,
-      state: {},
+      state: { showLoginDialog: false },
```

Comment:
had to change this because it was not allowing me to close the login dialog when in the `add` scholarships page

Reply:
i checked the console and when trying to close it, state would be `{}` and then it would get refreshed and state would again be `{showLoginDialog: true}`

my guess is that it was conflicting with how `ProtectedRoute` was setting `showLoginDialog`